### DISCUSSION TITLE
v48.0-alpha1

### DISCUSSION THREAD

**Discussion by westnordost** [streetcomplete/StreetComplete#4441] [Release Discussions]
<h2>M-M-M-Mega Update!</h2>
  <p>A few things came together... this update contains what would usually be in 3 or 4 releases! Have fun!</p>
<img src="https://user-images.githubusercontent.com/4661658/193340941-84d9544c-355c-4acb-ba01-2031e20357ef.png" width="300" align="right"/><img src="https://user-images.githubusercontent.com/4661658/193340924-492b148a-2f82-4fe5-8652-8e3d601281ed.png" width="300" align="right"/>
  <h3>1️⃣️️ Addresses Overlay</h3>
  <p>The most efficient way to map addresses on-site, even in places where no buildings are mapped yet!<br/>
  You can add addresses to buildings, edit existing ones and also add them at any new position, using the same sleek UI as for the address quests.</p>
  <h3>🛍️ Shops Overlay</h3>
  <p>Easily map and check what shops exist in your local shopping area!</p>
<br clear="right"/>
<br clear="left"/>
  <h3>🚀 HUGE performance improvements (#4079, #4125), by @Helium314</h3>
  <p>The app is now more responsive during normal use thanks to extensive caching, in particular, roughly:</p>
  <ul>
  <li>2x to 8x times faster display of quest pins and overlays</li>
  <li>2x to 3x times faster showing of quest details</li>
  <li>up to 2x times faster solving of a quests</li>
  </ul>
  <h3>New Quests</h3>
  <ul>
  <li><i>How is drinking water provided here?</i> (#4423, #4390), by @matkoniecz</li>
  <li><i>Which direction leads upwards here?</i>, asked for mountainbike tracks (#4385), by @matkoniecz</li>
  <li><i>What’s the identification number here?</i>, asked for emergency access points (#4384, #4386), by @mcliquid</li>
  </ul>
  <h3>Other enhancements</h3>
  <ul>
  <li>Redid UI for selecting the type of shop - better search, with icons from iD! (#3774)</li>
  <li>Redid UI for the housenumber and address quests to be consistent with the overlay</li>
  <li>Add links to documentation for tags changed as shown in undo dialog (#3442, #4419), by @matkoniecz</li>
  <li>The obstrusive downloading-view is gone, downloading and uploading is now shown in the upper left corner</li>
  <li>Other small enhancements and fixes (#4431, #4429, 4424, #4435...), thanks @matkoniecz, @mcliquid, @Helium314</li>
  </ul>

<hr /><em>This discussion was created from the release <a href='https://github.com/streetcomplete/StreetComplete/releases/tag/v48.0-alpha1'>v48.0-alpha1</a>.</em>

**Comment by deevroman** [comment]
Hmm, it seems that starting with this update, zooming with the +/- buttons periodically ceases to be smoothly animated. I can't reproduce it yet, because it appears and disappears randomly.

**Reply by westnordost** [reply]
This generally happens when the app is doing too much else on the UI thread. Displaying the new overlays puts somewhat on the load on that, so this may be the reason, if you have been trying those out.

**Reply by deevroman** [reply]
Overlays don't seem to affect this, because it occurs even when they are disabled.

**Reply by deevroman** [reply]
I'll clarify. Animation doesn't just get slow. It disappears completely. At the same time, the delay remains between switching zooms.

**Reply by deevroman** [reply]
https://user-images.githubusercontent.com/25708359/193355828-cb0cdd9e-f675-4e54-8a88-cfd17710446e.mp4

**Reply by westnordost** [reply]
I didn't do anything in that regard. As I said, maybe something is causing more work on the UI thread since v48.0-alpha1.

And you are sure you didn't have this with v47.0 at all?

**Reply by westnordost** [reply]
@Helium314 Maybe: the **first** time a place comes into view, it is put into cache and that may be a little slower than before? (And maybe this mistakenly is executed on the UI thread instead of IO thread?)

**Reply by Helium314** [reply]
I don't think so. The quests are already cached at this point, and map data should not be loaded at this point (without enabling overlays or opening quests).
And actually this does not look like the phone doing too much work... everything is quite smooth except for the button zoom.

**Reply by JLZIMMERMANN** [reply]
Very impresive et inspired improvements.
I tried to check shops : it seems not to let validate something already existing.

**Reply by westnordost** [reply]
I've noticed in the past that the zoom animation is cut off if there is another view animation running, like zooming to one quest after clicking on it but this doesn't seem to be the case here.

**Reply by matkoniecz** [reply]
> I'll clarify. Animation doesn't just get slow. It disappears completely. At the same time, the delay remains between switching zooms.

Have you maybe disabled animations in system settings of the phone?

**Reply by deevroman** [reply]
> And you are sure you didn't have this with v47.0 at all?

I tried it in the expert version 47.0. I also observe it there. Therefore, it probably did not appear in 48.0alpha. But perhaps the changes have made it more visible.

**Reply by deevroman** [reply]
> Have you maybe disabled animations in system settings of the phone?

It doesn't seem to have changed the settings. Then the animation would **never** be played.

**Reply by deevroman** [reply]
> I've noticed in the past that the zoom animation is cut off if there is another view animation running, like zooming to one quest after clicking on it but this doesn't seem to be the case here.

You seem to be right!

1. Make a swipe so that the map continues to move by inertia.
2. While the map is moving, press the +/- buttons
3. All subsequent clicks will be without animation.

Moreover, if you move the map without inertia, then at +/- the animation will not return. But if you move the map *with inertia* and *wait for the animation to finish*, the animation will return.

**Reply by mnalis** [reply]
I can confirm that I can reproduce with those instructions with both `v47.0` and `v48.0-alpha1`. I've also managed to reproduce it with one of the `v43.1` forks, so it seems that the bug have been here for some time, but maybe the app was faster so it didn't happen? Or maybe it's just luck... It **_is_** highly annoying when it happens. The workaround mentioned also works, though, when one knows it. Is this something that can be fixed in SC?

**Reply by mnalis** [reply]
So, may we open it as a new issue? Even if it is possibly TangremES bug, the effective workaround that works manually might be implemented in the SC code automatically?

**Comment by westnordost** [comment]
Why do you close discussions, Mateusz? Does this have any advantages? You know, everyone that left a comment on a discussion gets notified (by email) that it has been closed.

**Reply by mnalis** [reply]
> Does this have any advantages?

One advantage that I find useful is that one can look only at things that still need attention, e.g. via [is:unanswered](https://github.com/streetcomplete/StreetComplete/discussions?discussions_q=is%3Aunanswered) filter

**Reply by westnordost** [reply]
But this "unanswered" applies only to questions.

**Reply by riQQ** [reply]
[is:open](https://github.com/streetcomplete/StreetComplete/discussions?discussions_q=is%3Aopen) applies to all discussions.

**Reply by mnalis** [reply]
Yes, there is also `is:open` (and others), so one can filter on multiple of them, e.g. [is:unanswered is:open](https://github.com/streetcomplete/StreetComplete/discussions?discussions_q=is%3Aunanswered+is%3Aopen)

**Reply by matkoniecz** [reply]
> Why do you close discussions, Mateusz? Does this have any advantages?

Some discussions have actionable things to do, some are processed. Previously I either edited title (like in https://github.com/streetcomplete/StreetComplete/discussions/3429 ) or converted to Q&A.

But maybe approach "if discussion ever leads to anything actionable then someone should create an issue, all discussions should be assumed to be not actionable, issues opened in discussions are irrelevant" is preferable?

**Reply by westnordost** [reply]
> But maybe approach "if discussion ever leads to anything actionable then someone should create an issue, all discussions should be assumed to be not actionable, issues opened in discussions are irrelevant" is preferable?

IMO anything actionable that falls out of a discussion thread should either lead to a PR, a commit, an issue, or be ignored (because it is too minor to worry about that).
